server/src/main/java/org/opensearch/index/fielddata/plain/NonPruningSortedSetOrdinalsIndexFieldData.java

```diff
@@ -0,0 +1,340 @@
+            return delegate.getComparatorSource();
+        }
+
+        @Override
+        public String toString() {
+            return delegate.toString();
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            return delegate.equals(o);
+        }
+
+        @Override
+        public int hashCode() {
+            return delegate.hashCode();
+        }
+
+        @Override
+        public void setBytesComparator(Comparator<BytesRef> b) {
+            delegate.setBytesComparator(b);
+        }
+
+        @Override
+        public Comparator<BytesRef> getBytesComparator() {
+            return delegate.getBytesComparator();
+        }
+
+        @Override
+        public FieldComparator<?> getComparator(int numHits, Pruning pruning) {
+            return delegate.getComparator(numHits, pruning);
+        }
+
+        @Override
+        public SortField rewrite(IndexSearcher searcher) throws IOException {
+            return delegate.rewrite(searcher);
+        }
+
+        @Override
+        public boolean needsScores() {
+            return delegate.needsScores();
+        }
+
+        @Override
+        public IndexSorter getIndexSorter() {
+            return delegate.getIndexSorter();
+        }
+
+        @Deprecated
+        @Override
+        public void setOptimizeSortWithIndexedData(boolean optimizeSortWithIndexedData) {
+            delegate.setOptimizeSortWithIndexedData(optimizeSortWithIndexedData);
+        }
+
+        @Deprecated
+        @Override
+        public boolean getOptimizeSortWithIndexedData() {
+            return delegate.getOptimizeSortWithIndexedData();
+        }
+
+        @Deprecated
+        @Override
+        public void setOptimizeSortWithPoints(boolean optimizeSortWithPoints) {
+            delegate.setOptimizeSortWithPoints(optimizeSortWithPoints);
+        }
+
+        @Deprecated
+        @Override
+        public boolean getOptimizeSortWithPoints() {
+            return delegate.getOptimizeSortWithPoints();
+        }
+    }
+
+    /**
+     * {@link SortField} extension which disables pruning in the comparator.
+     *
+     * @opensearch.internal
+     */
+    public final class NonPruningSortField extends FilteredSortField {
+
+        private NonPruningSortField(SortField sortField) {
+            super(sortField);
+        }
+
+        public static Type readType(DataInput in) throws IOException {
+            return SortField.readType(in);
+        }
+
+        @Override
+        public Object getMissingValue() {
+            return delegate.getMissingValue();
+        }
+
+        @Override
+        public void setMissingValue(Object missingValue) {
+            delegate.setMissingValue(missingValue);
+        }
+
+        @Override
+        public String getField() {
+            return delegate.getField();
+        }
+
+        @Override
+        public Type getType() {
+            return delegate.getType();
+        }
```

Comment:
Oh -- you don't need to delegate all of these methods. That's what `FilteredSortField` is for.

The only two methods that you need to override are `getComparatorSource` and `getComparator`. Everything else will be inherited.